external-crates/move/crates/move-compiler/src/expansion/translate.rs

```diff
@@ -1099,6 +1136,97 @@ fn attribute(
     ))
 }
 
+pub(crate) fn error_attribute_value(
+    env: &mut CompilationEnv,
+    attributes: &E::Attributes,
+) -> Option<(Loc, u16)> {
+    let sp!(err_loc, err_attribute) = attributes.get_(&known_attributes::ErrorAttribute.into())?;
+    let E::Attribute_::Parameterized(sp!(_, name), inners) = &err_attribute else {
+        let msg = "Invalid #[error(..)] attribute";
+        let mut diag = diag!(Declarations::InvalidAttribute, (*err_loc, msg));
+        diag.add_note("Error attributes must be of the form #[error(code = <n>)], where 0 <= n <= 32767, e.g. #[error(code = 123)]");
+        env.add_diag(diag);
+        return None;
+    };
+
+    assert!(
+        name.as_str() == known_attributes::ErrorAttribute.name(),
+        "ICE error: unexpected attribute name for attribute #[error(..)]"
+    );
+
+    if inners.len() != 1 {
+        let msg = format!(
+            "#[error(..)] expects a single error code, found {}",
+            inners.len()
+        );
+
+        env.add_diag(diag!(Declarations::InvalidAttribute, (*err_loc, msg)));
+        return None;
+    }
+
+    let code_name = E::AttributeName_::Unknown(known_attributes::ErrorAttribute.code().into());
+
+    let mut invalid_attr = |loc| {
+        let msg = "Invalid code assignment used to assign error code in #[error(..)] attribute";
+        let mut diag = diag!(Declarations::InvalidAttribute, (loc, msg));
+        diag.add_note("Error attributes must be of the form #[error(code = <n>)], where 0 <= n <= 32767, e.g. #[error(code = 123)]");
```

Comment:
Nit: use `known_attributes::ErrorAttribute.error_code_max()`

Reply:
All going to go away now... :/